templates/etc/ansible/facts.d/apt.fact.j2

```diff
@@ -16,12 +16,12 @@
 {% endfor %}
 {% set apt__tpl_default_sources = [] %}
 {% for repo in apt__default_sources %}
-{%   if repo.uri|d() %}
+{%   if repo.uri|d() and repo.state|d('present') != 'absent' and (repo.distribution|d() and repo.distribution == apt__distribution) %}
 {%     for element in ([ repo.uri ] if repo.uri is string else repo.uri) %}
 {%       set _ = apt__tpl_default_sources.append(element) %}
 {%     endfor %}
 {%   endif %}
-{%   if repo.uris|d() %}
+{%   if repo.uris|d() and repo.state|d('present') != 'absent' and (repo.distribution|d() and repo.distribution == apt__distribution) %}
```

Comment:
I dunno, doesn't that mess up the `ifupdown__tpl_addresses` variable in the template? I remember that Jinja seems to apply all the "transformations" at once in parallel (yeah, weird, but it looked like that to me) and i had to go through hoops to avoid that - perhaps using `.copy()` would make the things simplier. Also, what's the context? And any issues with new `ifupdown` to report?